### PR TITLE
RAINCATCH-451: Update to use new simple store

### PR DESCRIPTION
**Pull request by JameelB**
## What
Updated server.js to add in an id field, required by the new raincatcher-simple-store module, to the result object upon creation.

## JIRA Ticket
https://issues.jboss.org/browse/RAINCATCH-451

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -6,4 +6,10 @@ var sync = require('fh-wfm-sync/lib/server')
 
 module.exports = function(mediator, app, mbaasApi) {
   sync.init(mediator, mbaasApi, config.datasetId, config.syncOptions);
+
+  // Add an id field required by the new simple store module to the result object to be created
+  mediator.subscribe('wfm:cloud:result:create', function(resultToCreate, ts) {
```
Comment (ghost) [inline lib/server.js]: whats `ts`?

**Comment by ghost** [comment]
@JameelB, @nialldonnellyfh think do we need some unit tests for this?

File: lib/server.js [hunk 2]
```diff
@@ -1,9 +1,48 @@
...
 
 
 module.exports = function(mediator, app, mbaasApi) {
   sync.init(mediator, mbaasApi, config.datasetId, config.syncOptions);
+
+  mediator.subscribe(config.topicPrefix + config.datasetId + ':create', function(resultToCreate, mediatorTopicIdentifier) {
+    // Add an id field required by the new simple store module to the result object to be created
+    resultToCreate.id = shortid.generate();
+
+    mediator.request(config.topicPrefix + 'data:' + config.datasetId + ':create', resultToCreate, {uid: resultToCreate.id})
```
Comment (nialldonnellyfh) [inline lib/server.js]: It would be good to have a small utility function to generate the topics. 

It would avoid concatenation errors for a single topic.

**Review by nialldonnellyfh — COMMENTED** [review]
Left a small comment around generating topics. 

It would be good to add some unit tests around this functionality.

**Pull request merged**